### ISSUE TITLE
Change handler should trigger update beatmap if needed.

### ISSUE THREAD

**Issue by andy840119** [karaoke-dev/karaoke#1900]
Call `PerformOnSelection()` in the `EditorBeatmap` will not trigger the `UpdateState()`.
Should manually call `Update(hitObject)` to trigger the re-processing the beatmap.

Because lyric change handler only focus on edit the one property, so guess should be ok to add the property like `TriggerUpdate` in the `HitObjectChangeHandler`.

Also:
- [x] Need to think about how to trigger the `UpdateState()` change in the non-hit-object change handler because there's no hit-object changed. -> guess use the `UpdateAllHitObjects()` i think.
- #1669 caused `beatmap.PerformOnSelection` not being called in the change handler(due to customized ruleset does not have the `TransactionalCommitComponent` injection). guess should use the temp way to fix that.

**Comment by andy840119** [comment]
Guess should make the `TriggerHitObjectUpdate(hitObject)` in the `HitObjectChangeHandler` to trigger the hit-object update.
Also add the `TriggerAllHitObjectUpdate()` in the `BeatmapChangeHandler` to force trigger re-run the beatmap processor.

Also, Have two ways to fix the time-tag dragging issue(#1669):
1. add extra params in the `PerformOnSelection ()`.
2. call extra property or flag before the `PerformOnSelection ()`, e.g.

```csharp
public TimeTag Shifting(TimeTag timeTag, ShiftingDirection direction, ShiftingType type)
{
    CheckExactlySelectedOneHitObject();

    // call this method first.
    NotTriggerSaveStateOnThisChange();

    TimeTag newTimeTag = null!;

    PerformOnSelection(lyric =>
    {
       // do something.
    });
}
```

**Comment by andy840119** [comment]
Implemented.
